venter/tests.py

```diff
@@ -1,7 +1,20 @@
+from django.core import mail
+from django.contrib.admin.sites import AdminSite
+
 from rest_framework.test import APITestCase
 from login.tests import get_new_user
-from venter.models import Complaints, TagUris, Comment, ComplaintMedia
+from venter.admin import ComplaintModelAdmin
+from venter.models import Complaints
+from venter.models import TagUris
+from venter.models import Comment
+from venter.models import ComplaintMedia
+from venter.models import Authorities
+
+class MockRequest(object):  # pylint: disable=R0205
```

Comment:
Why do we need this?

Reply:
I'm using mock request to pass requests to the admin actions while testing them because the admin action functions require a request to be passed to them, it's how they're written

Reply:
Ah, okay. You're needing to do this because you aren't testing admin actions correctly. Instead of calling the functions defined in the admin model directly, you should call them with the dummy client (which will then produce the request object for you). This might seem to be the same thing, but it actually isn't, since testing it through the dummy client also tests routing. [This](https://stackoverflow.com/a/29027215) answer might be relevant.